core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -356,8 +357,16 @@ class AdminManager(val config: KafkaConfig,
               createResponseConfig(allConfigs(config),
                 createBrokerConfigEntry(perBrokerConfig = true, includeSynonyms))
             else
-              throw new InvalidRequestException(s"Unexpected broker id, expected ${config.brokerId} or empty string, but received $resource.name")
+              throw new InvalidRequestException(s"Unexpected broker id, expected ${config.brokerId} or empty string, but received ${resource.name}")
 
+          case ConfigResource.Type.BROKER_LOGGER =>
+            if (resource.name == null || resource.name.isEmpty)
+              throw new InvalidRequestException("Broker id must not be empty")
```

Comment:
why not support this, like we do for broker configs?

Reply:
What does a lack of broker_id do for broker configs? It returns all the cluster-wide default configurations from ZK.
With broker loggers, we don't have a cluster-wide default configuration - only the locally-modified one made from the `log4j.properties`. Further, I don't think log4j gives us a way to get the default configs prior to any modification. I don't see what we can return here

Reply:
Sorry, it was my misunderstanding of what the empty broker_id returns.